tests/unit/preprocessor/_time/test_time.py

```diff
@@ -168,53 +166,115 @@ def test_extract_time_no_time(self):
+                                    standard_name='time',
+                                    units=Unit('days since 1950-01-01',
+                                               calendar='gregorian'))
+        cube = iris.cube.Cube(data, dim_coords_and_dims=[(time, 0)])
+        return cube
+
+    def test_clip_timerange_1_year(self):
+        """Test clip_timerange with 1 year."""
+        sliced = clip_timerange(self.cube, 1950, 1950)
         iris.coord_categorisation.add_month_number(sliced, 'time')
         iris.coord_categorisation.add_year(sliced, 'time')
         assert_array_equal(np.arange(1, 13, 1),
                            sliced.coord('month_number').points)
         assert_array_equal(np.full(12, 1950), sliced.coord('year').points)
 
-    def test_clip_start_end_year_3_years(self):
-        """Test clip_start_end_year with 3 years."""
-        sliced = clip_start_end_year(self.cube, 1949, 1951)
+    def test_clip_timerange_3_years(self):
+        """Test clip_timerange with 3 years."""
+        sliced = clip_timerange(self.cube, 1949, 1951)
         assert sliced == self.cube
 
-    def test_clip_start_end_year_no_slice(self):
-        """Test fail of clip_start_end_year."""
+    def test_clip_timerange_no_slice(self):
+        """Test fail of clip_timerange."""
         with self.assertRaises(ValueError) as ctx:
-            clip_start_end_year(self.cube, 2200, 2200)
+            clip_timerange(self.cube, 2200, 2200)
         msg = ("Time slice 2200-01-01 to 2201-01-01 is outside"
                " cube time bounds 1950-01-16 00:00:00 to 1951-12-07 00:00:00.")
         assert ctx.exception.args == (msg, )
+        with self.assertRaises(ValueError) as ctx:
+            clip_timerange(self.cube, 2200, 2200, '2200/2200')
+        assert ctx.exception.args == (msg, )
 
-    def test_clip_start_end_year_one_time(self):
-        """Test clip_start_end_year with one time step."""
+    def test_clip_timerange_one_time(self):
+        """Test clip_timerange with one time step."""
         cube = _create_sample_cube()
         cube.coord('time').guess_bounds()
         cube = cube.collapsed('time', iris.analysis.MEAN)
-        sliced = clip_start_end_year(cube, 1950, 1952)
+        sliced = clip_timerange(cube, 1950, 1952)
+        sliced_timerange = clip_timerange(cube, 1950, 1952, '1950/1952')
         assert_array_equal(np.array([360.]), sliced.coord('time').points)
+        assert_array_equal(np.array([360.]),
+                           sliced_timerange.coord('time').points)
 
-    def test_clip_start_end_year_no_time(self):
-        """Test clip_start_end_year with no time step."""
+    def test_clip_timerange_no_time(self):
+        """Test clip_timerange with no time step."""
         cube = _create_sample_cube()[0]
-        sliced = clip_start_end_year(cube, 1950, 1950)
+        sliced = clip_timerange(cube, 1950, 1950)
+        sliced_timerange = clip_timerange(cube, 1950, 1950, '1950/1950')
```

Comment:
The years are discarded if the timerange is set. Maybe you can make ths more explicit by passing None instead of a valid year

Reply:
During the preprocessing, all values are available in order to make the changes backward compatible. But sure I can set them to None here.